alibaba/GraphScope
# Sync with latest changes about arrow fragment in vineyard, and include several refactors/bugfixes.

## What do these changes do?

+ Fixes all compiler warnings about unused variable/compare-with-mismatch-types, etc.
+ Reduce the ad-hoc of manipulating metadata JSON during project and fragment transformation
+ Sync with the latest stream APIs in vineyard
+ Fixes a include path bug in CMakeLists.template to make sure the local build artifacts runnable without installation
+ Use `os.getcwd()` as the working directory of grape-engine, rather than the installation path of coordinator.
+ Enable use latest libgrape-lite/vineyard for CI
+ Enable colorful error/warning messages in CMakeLists.txt
